### PR TITLE
Add support for VS 2022 to the VSBuildV1

### PR DESCRIPTION
**Pull request by AnnaOpareva**
**Task name**: VSBuildV1

**Description**: Add support for VS 2022 to the VSBuildV1

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) [#15051](https://github.com/microsoft/azure-pipelines-tasks/issues/15051)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by EzzhevNikita** [comment]
@AnnaOpareva Please bump msbuildhelpers version in /Tasks/Common/MSBuildHelpers/package.json

File: Tasks/VSBuildV1/task.json [hunk 1]
```diff
@@ -158,6 +158,25 @@
...
+            "helpMarkDown": "Set custom version Visual Studio.",
+            "groupName": "advanced"
+        },
+        {
+            "name": "customVersion",
+            "type": "string",
+            "label": "Custom Version",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Set custom version Visual Studio.",
```
Comment (alexander-smolyakov) [inline Tasks/VSBuildV1/task.json:177]: I think it makes sense to add an example to the markdown helper for this input field

File: Tasks/VSBuildV1/Get-VSPath.ps1 [hunk 2]
```diff
@@ -27,4 +34,4 @@ function Get-VSPath {
     } finally {
         Trace-VstsLeavingInvocation $MyInvocation
     }
-}
+}
```
Comment (alexander-smolyakov) [inline Tasks/VSBuildV1/Get-VSPath.ps1:37]: Nit: EOL

File: Tasks/VSBuildV1/task.json [hunk 3]
```diff
@@ -158,6 +158,25 @@
...
             "helpMarkDown": "If true - enables default logger for msbuild",
             "groupName": "advanced"
+        },
+        {
+            "name": "isCustomVersion",
+            "type": "boolean",
+            "label": "Set custom version VS",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Set custom version Visual Studio.",
```
Comment (alexander-smolyakov) [inline Tasks/VSBuildV1/task.json:168]:
```suggestion
            "helpMarkDown": "Toggle to set a custom version of Visual Studio",
```

File: Tasks/Common/MSBuildHelpers/PathFunctions.ps1 [hunk 4]
```diff
@@ -19,7 +19,17 @@ function Get-MSBuildPath {
         # has regressed. When it is called for a version that is not found, the latest version
         # found is returned instead. Same for "16.0"
         [System.Reflection.Assembly]$msUtilities = $null
-        if (($Version -eq "16.0" -or !$Version) -and # !$Version indicates "latest"
+        if (($Version -eq "17.0" -or !$Version) -and # !$Version indicates "latest"
```
[the remark this answers is not in the document]
Comment (AnnaOpareva) [reply Tasks/Common/MSBuildHelpers/PathFunctions.ps1]: fixed

File: Tasks/VSBuildV1/Strings/resources.resjson/en-US/resources.resjson [hunk 5]
```diff
@@ -30,6 +30,10 @@
   "loc.input.help.logFileVerbosity": "Optional log file verbosity.",
   "loc.input.label.enableDefaultLogger": "Enable Default Logger",
   "loc.input.help.enableDefaultLogger": "If true - enables default logger for msbuild",
+  "loc.input.label.isCustomVersion": "Set custom version VS",
+  "loc.input.help.isCustomVersion": "Set custom version Visual Studio.",
+  "loc.input.label.customVersion": "Custom Version",
```
[the remark this answers is not in the document]
Comment (AnnaOpareva) [reply Tasks/VSBuildV1/Strings/resources.resjson/en-US/resources.resjson:33]: fixed

[the remark this answers is not in the document]
Comment (AnnaOpareva) [reply Tasks/VSBuildV1/Strings/resources.resjson/en-US/resources.resjson:33]: fixed

File: Tasks/Common/MSBuildHelpers/PathFunctions.ps1 [hunk 6]
```diff
@@ -12,23 +12,27 @@ function Get-MSBuildPath {
...
+        # found is returned instead. Same for "16.0" and "17.0"
         [System.Reflection.Assembly]$msUtilities = $null
-        if (($Version -eq "16.0" -or !$Version) -and # !$Version indicates "latest"
-            ($visualStudio16 = Get-VisualStudio 16) -and
-            $visualStudio16.installationPath) {
 
-            $msbuildUtilitiesPath = [System.IO.Path]::Combine($visualStudio16.installationPath, "MSBuild\Current\Bin\Microsoft.Build.Utilities.Core.dll")
+        if (($VersionNumber -ge 16 -or !$Version) -and # !$Version indicates "latest"
+            ($visualStudioNumber = Get-VisualStudio $VersionNumber) -and
+            $visualStudioNumber.installationPath) {
```
Comment (anatolybolshakov) [inline Tasks/Common/MSBuildHelpers/PathFunctions.ps1:27]: Nit: could it make sense to rename it to something like $specifiedStudio?

File: Tasks/VSBuildV1/Get-VSPath.ps1 [hunk 7]
```diff
@@ -6,10 +6,17 @@ function Get-VSPath {
 
     Trace-VstsEnteringInvocation $MyInvocation
     try {
+        if ($Version -eq "17.0" -and
+            ($instance = Get-VisualStudio 17) -and
```
Comment (anatolybolshakov) [inline Tasks/VSBuildV1/Get-VSPath.ps1:10]: Could we avoid duplicates here also?

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Review by EzzhevNikita — APPROVED** [review]

**Comment by EzzhevNikita** [comment]
tested this changes manually, found no issues

**Pull request merged**